firestore/src/ios/firestore_ios.cc

```diff
@@ -299,9 +322,74 @@ void Firestore::set_log_level(LogLevel log_level) {
                                                     : log_level);
 }
 
-void FirestoreInternal::SetClientLanguage(const std::string& language_token) {
+void FirestoreInternal::SetClientLanguage(const std::string &language_token) {
   api::Firestore::SetClientLanguage(language_token);
 }
 
+Future<LoadBundleTaskProgress> FirestoreInternal::LoadBundle(
+    const std::string &bundle) {
+  auto promise = promise_factory_.CreatePromise<LoadBundleTaskProgress>(
+      AsyncApi::kLoadBundle);
+  auto bundle_stream = absl::make_unique<util::ByteStreamCpp>(
+      absl::make_unique<std::stringstream>(std::stringstream(bundle)));
+  std::shared_ptr<api::LoadBundleTask> task =
+      firestore_core_->LoadBundle(std::move(bundle_stream));
+  task->Observe(
+      [promise, task](const api::LoadBundleTaskProgress &progress) mutable {
+        if (progress.state() == api::LoadBundleTaskState::kSuccess) {
+          promise.SetValue(ToApiProgress(progress));
+          task->RemoveAllObservers();
+        }
```

Comment:
nit: Use an "else if" here to be clearer about control flow. Also on line 367.

Reply:
Done.

Reply:
Done for real..